source/beamline/AMSlit.cpp

```diff
@@ -121,16 +120,20 @@ double AMSlit::centerValue() const
 	return result;
 }
 
-void AMSlit::addChildControl(AMControl *control)
+bool AMSlit::addChildControl(AMControl *control)
 {
-	if (allControls_->addControl(control))
-		AMControl::addChildControl(control);
+	if (AMControl::addChildControl(control)){
+		return allControls_->addControl(control);
+	}
+	return false;
 }
 
-void AMSlit::removeChildControl(AMControl *control)
+bool AMSlit::removeChildControl(AMControl *control)
 {
-	if (!allControls_->removeControl(control))
-		AMControl::removeChildControl(control);
+	if (AMControl::removeChildControl(control)){
+		return !allControls_->removeControl(control);
```

Comment:
should this return `allControls_->removeControl(control)` or `!  allControls_->removeControl(control)` ?